### PR TITLE
Add `CONTRIBUTING.md`

### PR DESCRIPTION
**Pull request by agileshaw**
This patch adds a contribution guide to provide information on how to become a contributor and how code changes should be submitted.

fixes: #211

### CODE REVIEW

**Review by rgildein — APPROVED** [review]
LGTM

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,72 @@
+# Contributing to Charmed OpenStack Upgrader (COU)
+
+Thank you for your interest in helping us with improving **COU**! We're open to
```
Comment (aieri) [inline CONTRIBUTING.md:3]:
```suggestion
Thank you for your interest in helping us improve **COU**! We're open to
```

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+    ```bash
+    cd docs/
+    make clean
+    make html
+    make spelling
+    make woke
+    make linkcheck
+    ``` 
+- Commit messages should be well-structured and provide a meaningful explanation
+- of the changes made
```
Comment (aieri) [inline CONTRIBUTING.md:58]:
```suggestion
  of the changes made
```

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+
+## Find issues to work on
+[GitHub Issues](https://github.com/canonical/charmed-openstack-upgrader/issues)
+is our central hub for bug tracking and issue management, with labels used to
+organize them into different categories. For new contributors, we recommend
+starting with issues labeled "good first issue." If you're interested in
+enhancing our documentation, you can filter issues using the "documentation"
+label to find issues specifically related to documentation improvement.
+  
+Once you decided which issue to work on, you can express your interest by posting
```
Comment (aieri) [inline CONTRIBUTING.md:69]:
```suggestion
Once you have decided which issue to work on, you can express your interest by posting
```

**Review by aieri — CHANGES_REQUESTED** [review]
Thank you, this looks good! I have requested a few minor tweaks.

**Review by gabrielcocenza — APPROVED** [review]

**Review by rgildein — APPROVED** [review]

**Review by aieri — APPROVED** [review]

**Pull request merged**